x/ccv/provider/keeper/misbehaviour.go

```diff
@@ -0,0 +1,154 @@
+package keeper
+
+import (
+	"github.com/cosmos/interchain-security/v2/x/ccv/provider/types"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
+	ibcclienttypes "github.com/cosmos/ibc-go/v4/modules/core/02-client/types"
+	ibctmtypes "github.com/cosmos/ibc-go/v4/modules/light-clients/07-tendermint/types"
+	tmtypes "github.com/tendermint/tendermint/types"
+)
+
+// HandleConsumerMisbehaviour checks if the given IBC misbehaviour corresponds to an equivocation light client attack,
+// and in this case, jails the Byzantine validators
+func (k Keeper) HandleConsumerMisbehaviour(ctx sdk.Context, misbehaviour ibctmtypes.Misbehaviour) error {
+	logger := k.Logger(ctx)
+
+	// Check that the misbehaviour is valid and that the client consensus states at trusted heights are within trusting period
+	if err := k.CheckMisbehaviour(ctx, misbehaviour); err != nil {
+		logger.Info("Misbehaviour rejected", err.Error())
+
+		return err
+	}
+
+	// Since the misbehaviour packet was received within the trusting period
+	// w.r.t to the trusted consensus states the infraction age
+	// isn't too old. see ibc-go/modules/light-clients/07-tendermint/types/misbehaviour_handle.go
+
+	// Get Byzantine validators from the conflicting headers
+	byzantineValidators, err := k.GetByzantineValidators(ctx, misbehaviour)
+	if err != nil {
+		return err
+	}
+
+	provAddrs := make([]types.ProviderConsAddress, len(byzantineValidators))
+
+	// slash, jail, and tombstone the Byzantine validators
+	for _, v := range byzantineValidators {
+		providerAddr := k.GetProviderAddrFromConsumerAddr(
+			ctx,
+			misbehaviour.Header1.Header.ChainID,
+			types.NewConsumerConsAddress(sdk.ConsAddress(v.Address.Bytes())),
+		)
+		err := k.SlashValidator(ctx, providerAddr)
+		if err != nil {
+			logger.Error("failed to slash validator: %s", err)
+			continue
+		}
+		err = k.JailAndTombstoneValidator(ctx, providerAddr)
+		if err != nil {
+			logger.Error("failed to jail or tombstone validator: %s", err)
+			continue
+		}
+
+		provAddrs = append(provAddrs, providerAddr)
+	}
+
+	logger.Info(
+		"confirmed equivocation light client attack",
+		"byzantine validators", byzantineValidators,
+		"byzantine validators that got slashes, jailed, and tombstoned", provAddrs,
```

Comment:
```suggestion
		"byzantine validators that got slashed, jailed, and tombstoned", provAddrs,
```

Reply:
Updated the logging to have each jailing/slashing error directly printed.
See [882e28942](https://github.com/cosmos/interchain-security/pull/1360/commits/882e2894271bb8841cd2253f3eb1582be4defe8d)